services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/IntersectionCriteria.java

```diff
@@ -0,0 +1,59 @@
+package us.dot.its.jpo.ode.api.accessors;
+
+import org.springframework.data.mongodb.core.query.Criteria;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.util.Date;
+
+public class IntersectionCriteria extends Criteria {
+
+    /**
+     * Build a query criteria object based on a time window
+     *
+     * @param fieldName        the db field to apply criteria to
+     * @param startEpochMillis the nullable start time of the window, in
+     *                         milliseconds since epoch
+     * @param endEpochMillis   the nullable end time of the window, in milliseconds
+     *                         since epoch
+     * @return the criteria object to use for querying
+     */
+    public IntersectionCriteria withinTimeWindow(
+            @Nonnull String fieldName,
+            @Nullable Long startEpochMillis,
+            @Nullable Long endEpochMillis,
+            @Nullable boolean formatAsString) {
+        if (startEpochMillis != null && endEpochMillis != null) {
+            this.and(fieldName)
+                    .gte(formatAsString == true ? Instant.ofEpochMilli(startEpochMillis).toString()
+                            : Date.from(Instant.ofEpochMilli(startEpochMillis)))
+                    .lte(formatAsString == true ? Instant.ofEpochMilli(endEpochMillis).toString()
+                            : Date.from(Instant.ofEpochMilli(endEpochMillis)));
+            return this;
+        } else if (startEpochMillis != null) {
+            this.and(fieldName).gte(formatAsString == true ? Instant.ofEpochMilli(startEpochMillis).toString()
+                    : Date.from(Instant.ofEpochMilli(startEpochMillis)));
+        } else if (endEpochMillis != null) {
+            this.and(fieldName).lte(formatAsString == true ? Instant.ofEpochMilli(endEpochMillis).toString()
+                    : Date.from(Instant.ofEpochMilli(endEpochMillis)));
+        }
+        return this;
```

Comment:
chore(blocking): remove the unnecessary `formatAsString == true` checks as `formatAsString` is already a boolean value.
```suggestion
        if (startEpochMillis != null && endEpochMillis != null) {
            this.and(fieldName)
                    .gte(formatAsString ? Instant.ofEpochMilli(startEpochMillis).toString()
                            : Date.from(Instant.ofEpochMilli(startEpochMillis)))
                    .lte(formatAsString ? Instant.ofEpochMilli(endEpochMillis).toString()
                            : Date.from(Instant.ofEpochMilli(endEpochMillis)));
        } else if (startEpochMillis != null) {
            this.and(fieldName).gte(formatAsString ? Instant.ofEpochMilli(startEpochMillis).toString()
                    : Date.from(Instant.ofEpochMilli(startEpochMillis)));
        } else if (endEpochMillis != null) {
            this.and(fieldName).lte(formatAsString ? Instant.ofEpochMilli(endEpochMillis).toString()
                    : Date.from(Instant.ofEpochMilli(endEpochMillis)));
        }
        return this;
```